server/src/types/formulas.ts

```diff
@@ -0,0 +1,48 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import * as yup from 'yup';
+
+export enum Formula {
+  Manual = 'MANUAL',
+  WeightedAverage = 'WEIGHTED_AVERAGE',
+}
+
+export enum Status {
+  Pass = 'pass',
+  Fail = 'fail',
```

Comment:
Probably, but I'm not completely sure if this enum is even necessary. It may be sufficient to just have `passed: boolean` in `GradingResult`.

@Gskartwii considered the possibility of adding other statuses in the future such as "grading in progress," but I don't know if that's relevant in this context. Maybe if it is relevant in another context, this may still be good to have in grade calculation to be consistent with other uses.